File: 0145-binary-tree-postorder-traversal/0145-binary-tree-postorder-traversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    void post_travs(TreeNode* root, vector<int>& vec){
        if (root == nullptr)
            return;
        post_travs(root->left, vec);
        post_travs(root->right, vec);
        vec.push_back(root->val);
    }

    vector<int> postorderTraversal(TreeNode* root) {
        vector<int> result;
        post_travs(root, result);
        return result;
    }
};

File: 0145-binary-tree-postorder-traversal/0145-binary-tree-postorder-traversal.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    def postorderTraversal(self, root):
        """
        :type root: TreeNode
        :rtype: List[int]
        """
        res = []
        def post_trav(root):
            if root is None:
                return
            post_trav(root.left)
            post_trav(root.right)
            res.append(root.val)
        post_trav(root)
        return res
        